src/themes/teams/components/Divider/dividerVariables.ts

```diff
@@ -1,25 +1,25 @@
+import * as _ from 'lodash'
+
 import { EmphasisColors, NaturalColors } from '../../../types'
 import { pxToRem } from '../../../../lib'
 
 export interface DividerVariables {
-  colors: EmphasisColors & NaturalColors
+  colors: Record<keyof (EmphasisColors & NaturalColors), string>
   dividerColor: string
   textColor: string
   textFontSize: string
   textLineHeight: string
-  primaryColor: string
   importantFontWeight: string
   dividerPadding: string
 }
 
 export default (siteVars: any): DividerVariables => {
   return {
-    colors: { ...siteVars.emphasisColors, ...siteVars.naturalColors },
+    colors: _.mapValues({ ...siteVars.emphasisColors, ...siteVars.naturalColors }, '500'),
```

Comment:
What is the magic `'500'` doing here? :)

Reply:
We have `colors` with following structure:
```js
const colors = {
  blue: { 500: 'blue' // other values },
  pink: { 500: 'pink'  },
}
```

I decided to reduce this structure with `mapValues` to as we using only single variant:
```js
const colors = {
  blue: 'blue',
  pink: 'pink',
}
```

This structure is more usable for overrides, too.
Yes, it looks like a magic number, but it is a simply color variant.

Reply:
We can do this:

```
const colorVariant = '500'

return {
  colors: _.mapValues({ ...siteVars.emphasisColors, ...siteVars.naturalColors }, colorVariant),
}
```

May be, this change will improve readability.

Reply:
We can save my comment here for another PR, but here are some thoughts.

***

Wondering if we should go this route, defining single color variable for every component.  Most components will implement the same variant of a color for the same purpose.  Example, a blue border will be the same shade in components that have blue borders.  Red text is similar where ever it appears.  The odds that a specific component needs to override a specific shade of color I think would be the exception and not the norm.

We might be able to produce a "scheme" for a component to implement.  Imagine picking colors along a color object like this:

```jsx
const primary = {
  50: '#F4F4FC',  // background
  100: '#E2E2F6', // shadow
  200: '#BDBDE6', // border
  300: '#8F90C1',
  400: '#6E70AE',
  500: '#6264A7', // text
  600: '#55578D',
  700: '#4A4C78',
  800: '#414265',
  900: '#33344A',
}
```
![image](https://user-images.githubusercontent.com/5067638/49560797-b0504b80-f8c8-11e8-8299-cd7f32e111a3.png)

Or, an inverted version of the same component using different variants from the same object:

```jsx
const primary = {
  50: '#F4F4FC',  // text
  100: '#E2E2F6',
  200: '#BDBDE6', // shadow
  300: '#8F90C1',
  400: '#6E70AE',
  500: '#6264A7', // background
  600: '#55578D',
  700: '#4A4C78', // border
  800: '#414265',
  900: '#33344A', 
}
```
![image](https://user-images.githubusercontent.com/5067638/49560933-6c117b00-f8c9-11e8-83e2-b0481d3b35e6.png)

### Divider

Applied to the Divider, I can imagine variables for the `borderColor` and a `textColor` since these are the relevant parts of the component and can have different colors, such as here:

![image](https://user-images.githubusercontent.com/5067638/49562965-96b40180-f8d2-11e8-98dc-d4d4dbb76aa5.png)

However, I think what is missing in this PR right now is that there is no indication of which variant the text color will use compared to the border variant.  We might want a way to choose a "scheme" for applying to a component or at least specify which variants will be used.

### Schemes

If we try the scheme route, we might look at many components in a UI and ask what the common color schemes are.  A scheme would includes colors for text, background, border, and shadow.

I can see defining a "primary" scheme or a "secondary" scheme, even "inverted" schemes in this way.  The scheme from the first Message looking component above would look something like this:

```jsx
const scheme = {
  foreground: primary[500],

  quiet: primary[400],         // range of "muted" foreground colors
  quieter primary[300],
  quietest: primary[200],

  background: primary[50],
  border: primary[200],
  shadow: primary[100]
}
```

These could be generated for any color, inversion, emphasis, and state like disabled.  They could also be adjusted for higher/lower contrast or brightness.

Reply:
Yes, it's a long discussion :) Not for this PR, that's why I removed `Divider` from examples of theming.

Reply:
@layershifter yep, please introduce the changes proposed in https://github.com/stardust-ui/react/pull/558#discussion_r it is much clear that way.